db/change_cache_test.go

```diff
@@ -1861,3 +1861,19 @@ func logChangesResponse(changes []*ChangeEntry) {
 	}
 
 }
+
+func TestMaxChannelCacheConfig(t *testing.T) {
+	channelCacheMaxChannels := []int{10, 50000, 100000}
+
+	for _, val := range channelCacheMaxChannels {
+		options := CacheOptions{
+			ChannelCacheOptions: ChannelCacheOptions{
+				MaxNumChannels: val,
+			},
+		}
+		db, testBucket := setupTestDBWithCacheOptions(t, options)
+		testBucket.Close()
+		tearDownTestDB(t, db)
+		assert.Equal(t, val, db.DatabaseContext.Options.CacheOptions.MaxNumChannels)
```

Comment:
Seems like reading values from the db after tearing down is perhaps over-reliant on the existing teardown not nilling out `db`, which could change in the future.

\Would be safer to do the assert before closing the bucket and tearing down the database.